vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -1145,8 +1183,14 @@ bool AdsImpl::ShowAd(
       << std::endl << "  url: " << notification_info->url
       << std::endl << "  uuid: " << notification_info->uuid;
 
-  notifications_->Add(*notification_info);
-  ads_client_->ShowNotification(std::move(notification_info));
+  notifications_->PushBack(*notification_info);
+
+#if defined(OS_ANDROID)
```

Comment:
This change should be platform agnostic, i.e. add a function to `notification_helper_{os}.cc` where {os} is `_win`, `_android` etc.

Reply:
Issue created for that https://github.com/brave/brave-browser/issues/6200